nca/NetworkConfig/NetworkConfigQueryRunner.py

```diff
@@ -117,57 +120,105 @@ def _execute_pair_configs_query(self, query_type, config1, config2, cmd_line_fla
 
     def _run_query_on_all_pairs(self):
-        res = 0
-        output = ''
-        queries_not_executed = 0
+        res, output, queries_not_executed = self._init_query_results_values()
         for config1 in self.configs_array:
             for config2 in self.configs_array:
                 if config1 != config2:
                     query_res, query_output, query_not_executed = self._execute_pair_configs_query(
                         self.query_name, self._get_config(config1), self._get_config(config2))
-                    res += query_res
-                    output += query_output + '\n'
-                    queries_not_executed += query_not_executed
+                    res, output, queries_not_executed = \
+                        self._update_query_results_after_one_iteration(res, query_res, output, query_output,
+                                                                       queries_not_executed, query_not_executed)
+
+        return self._return_final_query_results(res, output, queries_not_executed)
+
+    # following 3 def-s are for avoiding code repetition in previous 4 def-s
+    @staticmethod
+    def _init_query_results_values():
+        """
+        initializes the query variable results as following
+        res : 0 (the numerical result)
+        output : will be initialized to an empty list to be dumped into str later
+        queries not executed : 0 (the number of not executed queries)
+        rtype: int , list, int
+        """
+        return 0, [], 0
+
+    @staticmethod
+    def _update_query_results_after_one_iteration(result, iter_res, output, iter_output, num_not_exec, iter_not_exec):
+        """
+        gets the query result from one iteration on the configs_array, and updates and returns the general
+        results variables as following:
+        1- adds the iteration numerical result to the existing numerical result
+        2. append the iteration output to the existing one
+        3. adds the iter_not_exec to the existing number of not executed queries
+        :param int result: the numerical result from running previous iterations
+        :param int iter_res: the numerical result from running this iteration
+        :param list output: the output from previous iterations
+        :param Union[str, dict] iter_output: the query output from last iteration
+        a dict in case of yaml, json output format , otherwise str
+        :param int num_not_exec: the number of not executed queries from previous iterations
+        :param int iter_not_exec: the query_not_executed result from last iteration (0/1)
+        rtype: int , list, int
+        """
+        result += iter_res
+        output.append(iter_output)
+        num_not_exec += iter_not_exec
+        return result, output, num_not_exec
+
+    def _return_final_query_results(self, res, output, queries_not_executed):
+        """
+        gets the final query results after running it on all iterations of configs_array
```

Comment:
These methods are used for a repeated set of objects, both as input and output (`res, output, queries_not_executed)`.
I think it might be easier to encapsulate this set of objects in another class, that can support the required operations (e.g. method such as 'add'.)